src/API/Reports/Coupons/Stats/DataStore.php

```diff
@@ -78,6 +78,15 @@ protected function update_sql_query_params( $query_args, &$totals_params, &$inte
 			$coupons_where_clause .= " AND ( {$order_status_filter} )";
 		}
 
+		/**
+		 * Filter the coupon stats WHERE clause before retrieving the coupon stats data.
+		 *
+		 * Allows modification of the coupon stats select criteria.
+		 *
+		 * @param string $coupons_where_clause The generated WHERE clause.
+		 * @param array  $query_args           The original arguments for the request.
+		 */
+		$coupons_where_clause           = apply_filters( 'wc_admin_coupon_stats_where_clause', $coupons_where_clause, $query_args );
```

Comment:
I've asked this on another PR as well, but what do you think about passing the table name to the filter as well?